src/auth/auth.service.ts

```
@@ -142,4 +141,13 @@ export class AuthService {
   async getUserPortfolios(userId: string): Promise<Portfolio[]> {
     return this.portfolioRepository.find({ where: { user: { id: userId } }, order: { createdAt: 'DESC' } });
   }
-}
+
+  async findByEmail(email: string): Promise<User | null> {
+    return await this.userRepository.findOne({ where: { email } });
+  }
+  
```

ESLint

[error] Delete `··`

(prettier/prettier)